perfhud/src/ts/PerfHud.tsx

```diff
@@ -82,40 +85,73 @@ class PerfHud extends React.Component<PerfHudProps, PerfHudState> {
     });
   }
   
-  selectChanged = (evt: any) => {
-    let current = this.state.pages.filter((p: PerfPage) => p.id == parseInt(evt.target.value, 10))[0];
-      if (typeof current == 'undefined') {
-        current = this.state.pages[0];
-      }
+  changePage = (id:number) => {
+    let current = this.state.pages[id];
+    this.changePerfPage(current);
+  }
+
+  changePerfPage = (page:PerfPage) => {
+      this.setState({
+        pages: this.state.pages,
+        minimized: false,
+        currentPage: page
+    });
+  }
+  setMinimized = ( mini: boolean ) => {
     this.setState({
       pages: this.state.pages,
-      minimized: this.state.minimized,
-      currentPage: current
+      minimized: mini,
+      currentPage: this.state.currentPage
     });
   }
```

Comment:
you can just do this.setState({ minimized: mini } as any) here if you don't want to do it long hand